### PR TITLE
fix: execute "compact-series-file" at startup (1.x)

### LINKED ISSUE #24256: Compact Series File as Startup Script (1.X)
When the database is online for long periods and many deletes occur, the series files grow. Instead of deleting the series outright, a tombstone is written instead. We will execute `influx_inspect buildtsi -compact-series-file` at startup which should help alleviate this problem.

[-compact-series-file documentation](https://docs.influxdata.com/enterprise_influxdb/v1.10/tools/influx_inspect/#--compact-series-file-)
```
Does not rebuild the index. Compacts the existing series file, including offline series. Iterates series in
each segment and rewrites non-tombstoned series in the index to a new .tmp file next to the segment.
Once all segments are converted, the temporary files overwrite the original segments.
```

### PR DESCRIPTION
**Pull request by bnpfeife**
- Closes #24256

(From Issue Description:)

When the database is online for long periods and many deletes occur, the series files grow. Instead of deleting the series outright, a tombstone is written instead. We will execute influx_inspect buildtsi -compact-series-file at startup which should help alleviate this problem.

### CODE REVIEW

File: scripts/influxd-systemd-start.sh [hunk 1]
```diff
@@ -1,5 +1,51 @@
...
+        then
+                # Once within the target section, search the key-value pairs for the
+                # desired key. Currently, this only supports string values. Since
+                # this is used only for wal-dir and data-dir, this should be okay.
+                if [[ "${line}" =~ ^[[:space:]]*${target}[[:space:]]*=[[:space:]]\"(.*)\" ]]
+                then
+                    echo "${BASH_REMATCH[1]}" ; return
+                fi
+        fi
+    done <<< "$(influxd config -config /etc/influxdb/influxdb.conf 2>/dev/null)"
```
Comment (davidby-influx) [inline scripts/influxd-systemd-start.sh]: Should `/etc/influxdb/influxdb.conf` be a constant/parameter that we use here and in the actual command line, so they can't get out of sync?

**Review by davidby-influx — CHANGES_REQUESTED** [review]
IS the config file location hard-coded here? Can it come from `INFLUXD_OPTS`

File: scripts/influxd-systemd-start.sh (hunk 1 above)
Comment (bnpfeife) [reply scripts/influxd-systemd-start.sh]: The config is hard-coded on `master-1.x`: [`scripts/influxd-systemd-start.sh`](https://github.com/influxdata/influxdb/blob/53856cdaae63a699946b388ffbba5bb56784ba51/scripts/influxd-systemd-start.sh#L3). I am unsure what supplying another `-config <path>` would do (override vs ignore vs error), but the behavior should be consistent with the call that starts the service.

Comment (bnpfeife) [reply scripts/influxd-systemd-start.sh]: I tested it, if `-config` is supplied via `INFLUXD_OPTS`, the first will be overridden in favor of `INFLUXD_OPTS`.

File: scripts/influxd-systemd-start.sh [hunk 2]
```diff
@@ -1,5 +1,51 @@
...
+then
+    # If this daemon is configured to run as root, influx_inspect hangs
+    # waiting for confirmation before executing. Supplying "yes" allows
+    # the service to continue without interruption.
+    yes | /usr/bin/influx_inspect buildtsi -compact-series-file \
+        -datadir "${DATA_DIR}"                                  \
+        -waldir  "${WAL_DIR}"
+fi
+
 /usr/bin/influxd -config /etc/influxdb/influxdb.conf $INFLUXD_OPTS &
```
Comment (davidby-influx) [inline scripts/influxd-systemd-start.sh:49]: Still think you should unify config file location on lines 34 and 49 into a single parameter/constant.

**Review by davidby-influx — CHANGES_REQUESTED** [review]
One change.  LMK if I am misunderstanding something.

**Review by davidby-influx — APPROVED** [review]
LGTM

**Pull request merged**